ksql-engine/src/main/java/io/confluent/ksql/query/id/SpecificQueryIdGenerator.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.query.id;
+
+import java.util.NoSuchElementException;
+
+/**
+ * Returns a specific query Id identifier based on what's set. Only returns each set Id once and
+ * will throw an exception if getNextId() is called twice without update.
+ */
+public class SpecificQueryIdGenerator implements QueryIdGenerator {
+
+  private long nextId;
+  private boolean alreadyUsed;
+
+  public SpecificQueryIdGenerator() {
+    this.nextId = 0L;
+    this.alreadyUsed = true;
+  }
+
+  public void setNextId(final long nextId) {
+    alreadyUsed = false;
+    this.nextId = nextId;
+  }
+
+  @Override
+  public long peekNext() {
+    return alreadyUsed ? nextId + 1 : nextId;
+  }
+
+  @Override
+  public String getNext() {
+    if (alreadyUsed) {
+      throw new NoSuchElementException("QueryIdGenerator has not been updated with new offset");
```

Comment:
let's throw a `KsqlServerException` here because I don't think this should ever happen, and if so we should throw a 5xx error

Reply:
This code is in generator, so would we want to introduce the notion of the server into engine, or should we keep it decoupled? I guess it could makes sense since this method involves calling setNextId() from the server in order for this not to throw an error.